pycalphad/io/cs_dat.py

```
@@ -1159,6 +1162,999 @@ def parse_cs_dat(instring):
+                            print(f'List of known endmembers: {endmember_names}')
+                            print('Write aborted')
+                            return
+                elif phase_model == 'SUBLM':
+                    # Get indices of participating constituents in phase (order of printed endmembers)
+                    indices = []
+                    for sublattice in tc_constituents:
+                        for species in sublattice:
+                            try:
+                                indices.append(1 + flat_constituents.index(species.name))
+                            except ValueError:
+                                print(f'Can\'t find constituent {species.name}')
+                # Now write excess magnetic terms for current constituent_set
+                output += f'{len(indices):4}\n'
+                # TODO: Get order properly if possible
+                order = 1
+                output += f'{"".join([f"{ind:4}" for ind in indices])}{order:4}\n'
+                # Check if magnetic parameters are provided as piecewise
+                if type(tc_value) == type(Piecewise([0,True])):
+                    print(f'Piecewise Curie temperature in species {name} of phase {phase_name} not permitted.')
+                    print(f'Value from first temperature interval will be used.')
+                    print(f'Check original expression to determine if this is acceptable:')
+                    print(tc_value)
+                    print()
+                    tc_value = float(tc_value.args[0])
+                if type(bmagn_value) == type(Piecewise([0,True])):
+                    print(f'Piecewise magnetic moment in species {name} of phase {phase_name} not permitted.')
+                    print(f'Value from first temperature interval will be used.')
+                    print(f'Check original expression to determine if this is acceptable:')
+                    print(bmagn_value)
+                    print()
```

This should warn or raise instead of printing